tensorflow/core/kernels/crop_and_resize_op.cc

```diff
@@ -189,11 +187,12 @@ class CropAndResizeOp : public AsyncOpKernel {
   float extrapolation_value_;
 };
 
-// Partial specialization of CropAndResize functor for a CPUDevice.
+// Partial specialization of CropAndResize functor for a CPUDevice and
+// SYCLDevice.
 namespace functor {
-template <typename T>
-struct CropAndResize<CPUDevice, T> {
-  bool operator()(const CPUDevice& d, typename TTypes<T, 4>::ConstTensor image,
+template <typename Device, typename T>
+struct CropAndResizeNonCuda {
```

Comment:
All these `NonCuda` make me uncomfortable.

Is it not possible to have specialization in another way?